languagetool-server/src/main/java/org/languagetool/server/LanguageToolHttpHandler.java

```diff
@@ -123,7 +123,7 @@ public void handle(HttpExchange httpExchange) throws IOException {
         // healthcheck should come before other limit checks (requests per time etc.), to be sure it works: 
         String pathWithoutVersion = path.substring("/v2/".length());
         if (pathWithoutVersion.equals("healthcheck")) {
-          if (workQueueFull(httpExchange, parameters, "Healthcheck failed: There are currently too many parallel requests.")) {
+          if (workQueueFull(httpExchange, parameters, "Healthcheck failed: There are currently too many parallel requests.") || textCheckerV2.checkerQueueAlmostFull()) {
```

Comment:
workQueueFull actually sends an error, the added case doesn't
refactor that part out and call it in the conditional so that we actually send a response